finite() has been deprecated since 10.9 and has been removed in the
arm64 macOS SDK. Use the standard C99 isfinite() instead.

<https://trac.macports.org/ticket/69124>

--- hotspot/src/share/vm/utilities/globalDefinitions_gcc.hpp.orig	2023-04-18 14:55:08
+++ hotspot/src/share/vm/utilities/globalDefinitions_gcc.hpp	2024-01-19 00:03:57
@@ -249,8 +249,8 @@ inline int g_isfinite(jfloat  f)                 { ret
 
 // Checking for finiteness
 
-inline int g_isfinite(jfloat  f)                 { return finite(f); }
-inline int g_isfinite(jdouble f)                 { return finite(f); }
+inline int g_isfinite(jfloat  f)                 { return isfinite(f); }
+inline int g_isfinite(jdouble f)                 { return isfinite(f); }
 
 
 // Wide characters
